subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/SharedJavaInstallationRegistryIntegrationTest.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+
+class SharedJavaInstallationRegistryIntegrationTest extends AbstractIntegrationSpec {
+
+    def "installation registry empty without environment setup"() {
+        buildFile << """
+            import org.gradle.jvm.toolchain.internal.SharedJavaInstallationRegistry;
+            import javax.inject.Inject
+
+            abstract class ShowPlugin implements Plugin<Project> {
+                @Inject
+                abstract SharedJavaInstallationRegistry getRegistry()
+
+                void apply(Project project) {
+                    project.tasks.register("show") {
+                        println "installations:" + registry.listInstallations()
```

Comment:
Since this is going to be a core feature, I believe we should have a task that prints the known java installations (or toolchains).
This can of course be added later in a follow-up PR.

Reply:
Good idea! But I think this would need to come a little later once we have the plumbing between `JavaInstallation` and the raw locations (which may actually be an awesome driver to start this). I'd assume you want something like

`./gradlew toolchains`
```
+ toolchains
    + /home/foo/bar
          | Type:       Oracle JDK
          | Version:    8.1.2
          | Vendor:     Oracle
          | Tools:
             ⤷ JavaCompiler
             ⤷ JavaLauncher
             ⤷ GenericTool(jlink)
```